lib/common/schemas.c

```diff
@@ -660,7 +658,7 @@ validate_xml_verbose(const xmlNode *xml_blob)
 
     dump_file(filename);
 
-    doc = xmlParseFile(filename);
+    doc = xmlReadFile(filename, NULL, 0);
```

Comment:
I think we're good here.  xmlReadFile doesn't say anything about it, but it uses xmlCreateURLParserCtxt internally, which does mention support for compressed files.

Reply:
Last parameter of xmlReadFile is options and I guess that is where XML_PARSE_DTDLOAD might go.
xmlRelaxNGParserCtxt seems not to have anything to do with these defaults but xmlReadFile creates a xmlParserCtxt.

Reply:
In fact options for xmlReadFile should probably be XML_PARSE_DTDLOAD | XML_PARSE_NOENT

Reply:
xmlLoadExtDtdDefaultValue and xmlSubstituteEntitiesDefault were first added by 69aef651d9 back in 2008.  They've been moved around by various commits since then.  It seems unlikely there's anyone around who can remember why they might have been relevant.

Reply:
From what I've seen in the libxml-code I guess it doesn't really matter where they are as long as that happens before xmlParseFile.

Reply:
> In fact options for xmlReadFile should probably be XML_PARSE_DTDLOAD | XML_PARSE_NOENT

`xmlReadFile()` can accept those options. I didn't use them here, because the legacy options weren't being set prior to calling `xmlParseFile()`.